### ISSUE TITLE
G1Xon identifier code

### ISSUE THREAD

**Issue by AndersonAOS** [shooking/ZoomPedalFun#26]
F0 52 00 XX 50 F7

??  – G1XON
4F = B3
58 - MS-50G
59 – G3X 
5A - G3
5B – G5
5F MS-60B
61 - MS-70CDR
63 -  G1ON
65 – B1on
66 – B1XON
6E – G5n

**Comment by AndersonAOS** [comment]
64 ??
if you have the identifiers of other models, post here.

**Comment by shooking** [comment]
Hi there. I have put them in wiki for pedals I have.
I think you correctly got the others. The GCE-3 and the B1XFour and friends seem to be 6E.

**Comment by AndersonAOS** [comment]
And the G1xon? Would that really be 64?

**Comment by shooking** [comment]
I had a genius idea of looking for some online patches - surely that WOULD show the hex?
Seems most folks share screen dumps ... computers are too much work .. doh.
But if you can find some patch dumps that might be a great confirmation?
For example - from a B1XFour
```
C:\Users\shooking1\Documents\Software\ZoomPedalFun\B1XFour\DerivedData\Patches>hexdump -C patch_00.bin
00000000  f0 52 00 6e 45 00 00 00  00 00 00 78 05 00 50 54  |.R.nE......x..PT|
00000010  43 46 54 01 00 00 00 01  00 00 00 06 00 10 00 00  |CFT.............|
00000020  00 00 00 00 00 00 00 00  00 00 00 53 61 00 6e 73  |...........Sa.ns|
```
So the patch itself has to have the device ID. If we can find some online patches we can verify the ID?

Anyhow I found this tonelib forum error message that sheds some light?

https://tonelib.net/forums/threads/transfer-g1on-patches-to-g1xon.9647/

![image](https://user-images.githubusercontent.com/22934103/214437050-9ce39f25-d07b-4feb-b63d-a13ec8e733b7.png)

So I would imaging g1xon = 0x63 and g1on = 0x64?

Now - dont forget sysextones ... https://github.com/SysExTones/g1on/blob/master/g1on/CONSTANTS.py#:~:text=ZOOM_DEVICE_ID_G1ON%20%3D%200x63
So maybe dude on the forum got it wrong way around? I only have a B1On which is for sure 0x61.
I will try to look more deeply later in the week - hassle me if I forget :-)

**Comment by shooking** [comment]
I took a quick look at some of the bastardized ToneLib patches.
B1XFour - 1st byte 6E
G1Xon - 1st byte 64 (d)
Now might just be coincidence - or the dude strips off the initial sysex to save space?
You could look at a MS70 to see if that is 61 - then the gun would for sure be smoking?

**Comment by AndersonAOS** [comment]
Excellent. I have the g1on and confirm that it is 63 however Never had at hand the

**Comment by AndersonAOS** [comment]
There is a midi command of "question" to which the crankset answers it is possible to know the identification code, but I do not have access to g1xon at the moment

**Comment by shooking** [comment]
Yeah it is on the wiki pages I wrote for pedals I have. You have seen the wiki right? If you want to contribute any pages that would be cool.

Tonight I finished creating a prototype PCB for my B1XFour remote control device. It will make my B1XFour into a G5n like control surface plus ability to inject (possibly midi over bluetooth) into the ESP32 to control FS1, 2, Tuner on, Fax category on, Fx1 to 5 on/off, settings, mem stomp, rhythm, looper and the p1 to 4 up/down via footswitch. I also added BT page Turner to it. I made a video about it on my YT channel. The idea seems applicable to other pedals but would need a different 25 pin cable wired into box.

**Comment by dkts2000** [comment]
i have a g1xon. id is 64
